tests/testerina-integration-test/src/test/resources/project-based-tests/data-providers/tests/new-data-provider-tests.bal

```diff
@@ -0,0 +1,118 @@
+// under the License.
+
+import ballerina/test;
+
+@test:Config {
+    dataProvider: dataGen
+}
+function intDataProviderTest(int value1, int value2, int result1) returns error? {
+    test:assertEquals(value1 + value2, result1, msg = "The sum is not correct");
+}
+
+@test:Config {
+    dataProvider: dataGen2
+}
+function fruitsDataProviderTest(int value1, int value2, string fruit) returns error? {
+    test:assertEquals(value1, value2, msg = "The sum is not correct");
+    test:assertEquals(fruit.length(), 6);
+}
+
+@test:Config{
+    dataProvider: dataGen3
+}
+function jsonDataProviderTest (json json1, json json2, json json3) {
+    json a = {"a": "a"};
+    json b = {"b": "b"};
+    json c = {"c": "c"};
+    test:assertEquals(json1, a, msg = "json data provider failed");
+    test:assertEquals(json2, b, msg = "json data provider failed");
+    test:assertEquals(json3, c, msg = "json data provider failed");
+}
+
+@test:Config {
+    dataProvider: dataGen4
+}
+function errorDataProviderTest(int value1, int value2, int result1) returns error? {
+    test:assertEquals(value1 + value2, result1, msg = "The sum is not correct");
+}
+
+@test:Config {
+    dataProvider: dataGen5
+}
+function invalidDataProviderTest(int value1, int value2, int result1) returns error? {
+    test:assertEquals(value1 + value2, result1, msg = "The sum is not correct");
+}
+
+@test:Config {
+    dataProvider: dataGen6
+}
+function testFunction1(int value1, int value2, int result1) returns error? {
+    test:assertEquals(value1 + value2, result1, msg = "The sum is not correct");
+}
+
+@test:Config {
+    dataProvider: dataGen7
+}
+function intArrayDataProviderTest(int value1, int value2, int result1) returns error? {
+    test:assertEquals(value1 + value2, result1, msg = "The sum is not correct");
+}
+
+function dataGen() returns map<[int, int, int]>|error {
+    map<[int, int, int]> dataSet = {
+        "Case1": [1, 2, 4],
+        "Case2": [10, 20, 31],
+        "Case3": [5, 6, 11]
+    };
+    return dataSet;
+}
+
+function dataGen2() returns map<[int, int, string]>|error {
+    map<[int, int, string]> dataSet = {
+        "banana": [10, 10, "banana"],
+        "cherry": [5, 5, "cherry"]
+    };
+    return dataSet;
+}
+
+function dataGen3() returns map<[json,json, json]> {
+    map<[json, json, json]> dataSet = {
+        "json1": [{"a": "a"}, {"b": "b"}, {"c": "c"}],
+        "json2": [{"a": "a"}, {"b": "b"}, {"c": "c"}]
+    };
+    return dataSet;
+}
+
+function dataGen4() returns map<[string]>|error{
+    return error("Error occurred while generating data set.");
+}
+
+function dataGen5() returns (string) {
+    return "this is invalid";
+}
+
+function dataGen6() returns map<[int, int, int]>|error {
+    map<[int, int, int]> dataSet = {
+        "CaseNew1": [1, 2, 3],
+        "CaseNew2": [10, 20, 30],
+        "Case3": [5, 6, 11]
+    };
+    return dataSet;
+}
+
+function dataGen7() returns int[][] {
+    return [[1, 2, 4], [10, 20, 31], [5, 6, 11]];
+}
```

Comment:
Can we add a test case with `#` in the key

Reply:
Fixed with 6603263f96dfe52dcbb0ee08e3725684f3261d0d